### PR TITLE
Add asdf.transformation

### PR DESCRIPTION
**Pull request by 9il**
#### TODO
- [x] Object-tree representation
- [x] Add
- [x] Remove
- [x] 'Add' difference
- [x] 'Remove' difference
- [x] documentation
- [x] unittests

### CODE REVIEW

File: source/asdf/transform.d [hunk 1]
```diff
@@ -0,0 +1,369 @@
+/++
+Mutable ASDF data structure.
+The representation can be used to compute a difference between JSON object-trees.
+
+Copyright: Tamedia Digital, 2016
+
+Authors: Ilya Yaroshenko
+
+License: MIT
```
Comment (yannick) [inline source/asdf/transform.d]: should be boost ;)

File: source/asdf/transform.d [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
++/
+module asdf.transformation;
+
+import asdf.asdf;
+import asdf.serialization;
+
+/++
+Object-tree structure for mutable Asdf representation.
```
Comment (yannick) [inline source/asdf/transform.d]: please document what this is for, when it should/can be used and for what reasons.
also add an example

Comment (9il) [inline source/asdf/transform.d]: Description added. There are few unittests for this in the structure declarations (for various methods).

**Comment by yannick** [comment]
@9il please quickly check again and then merge if it does not affect any functionality

**Pull request merged**